conversion/indico_conversion/codimd.py

```diff
@@ -0,0 +1,335 @@
+    """Build an ORM query which gets all attachments."""
+    contrib_event = db.aliased(Event)
+    contrib_session = db.aliased(Session)
+    subcontrib_contrib = db.aliased(Contribution)
+    subcontrib_session = db.aliased(Session)
+    subcontrib_event = db.aliased(Event)
+    session_event = db.aliased(Event)
+
+    attachment_filter = db.and_(
+        ~Attachment.is_deleted,
+        ~AttachmentFolder.is_deleted,
+        db.or_(
+            AttachmentFolder.link_type != LinkType.event,
+            ~Event.is_deleted,
+        ),
+        db.or_(
+            AttachmentFolder.link_type != LinkType.contribution, ~Contribution.is_deleted & ~contrib_event.is_deleted
+        ),
+        db.or_(
+            AttachmentFolder.link_type != LinkType.subcontribution,
+            db.and_(~SubContribution.is_deleted, ~subcontrib_contrib.is_deleted, ~subcontrib_event.is_deleted),
+        ),
+        db.or_(AttachmentFolder.link_type != LinkType.session, ~Session.is_deleted & ~session_event.is_deleted),
+    )
+
+    return (
+        (
+            Attachment.query.join(Attachment.folder)
+            .outerjoin(AttachmentFolder.linked_event)
+            .outerjoin(AttachmentFolder.contribution)
+            .outerjoin(Contribution.event.of_type(contrib_event))
+            .outerjoin(Contribution.session.of_type(contrib_session))
+            .outerjoin(AttachmentFolder.subcontribution)
+            .outerjoin(SubContribution.contribution.of_type(subcontrib_contrib))
+            .outerjoin(subcontrib_contrib.event.of_type(subcontrib_event))
+            .outerjoin(subcontrib_contrib.session.of_type(subcontrib_session))
+            .outerjoin(AttachmentFolder.session)
+            .outerjoin(Session.event.of_type(session_event))
+            .filter(attachment_filter)
+            .filter(AttachmentFolder.link_type != LinkType.category)
+        ),
+        session_event,
+        contrib_event,
+        subcontrib_event,
+    )
+
+
+def parse_codimd_url(url: str) -> tuple[str, bool] | None:
+    """Parse a CodiMD URL and return the document ID and whether it is a published link.
+
+    :param url: The URL to parse
+    :return: A tuple of (document_id, is_published) if the URL is valid, None otherwise.
+    """
+    from indico_conversion.plugin import ConversionPlugin
+
+    if not url or not isinstance(url, str):
+        return None
+
+    codimd_base_url = ConversionPlugin.settings.get('codimd_base_url')
+    if not codimd_base_url:
+        return None
+
+    # First check, to see if it's worth trying to parse the URL at all
+    if not url.startswith(codimd_base_url.rstrip('/')):
+        return None
+
+    try:
+        parsed = urlparse(url)
+        path = parsed.path.strip('/')
+
+        # Check if the base path is present in the URL, and if it is, remove it
+        base_path = urlparse(codimd_base_url).path.strip('/')
+        if base_path and path.startswith(base_path):
+            path = path[len(base_path) :].lstrip('/')
```

Comment:
ruff lets you do that?! (looks super weird IMO)


```suggestion
            path = path[len(base_path):].lstrip('/')
```